### PR TITLE
Performance Enhancements

### PR DESCRIPTION
**Pull request by mwm-deriv**
This changeset improves the performance of `Date::Utility` with respect to object creation.

Objects are now created in about a third (~4μs) of the time used on the current `master` branch (~12μs).  This performance benefit accrues to all `Date::Utility` consumers without any code changes.

The implementations changes are largely:

- Use constant regexps which are only compiled once for faster comparisons
- Remove the use of `Tie::Cache::LRU`, the overhead of which was higher than simply creating new objects
- Switch from Moose to Moo for faster MOP object building

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,412 @@
+This software is copyright (c) 2022 by Binary Services Ltd.
```
Comment (shaydo-deriv) [inline LICENSE]: @mwm-deriv I see that the LICENSE is present in distribution on CPAN. I expect that this file is added by distzilla during release process, so it needs not to be added

File: lib/Date/Utility.pm [hunk 2]
```diff
@@ -407,51 +401,34 @@ sub _build_is_a_weekday {
...
-        }
-    } elsif ($params_ref =~ $EPOCH_RE) {
-        $new_params->{epoch} = $params_ref;
-    } else {
-        $new_params = _parse_datetime_param($params_ref);
-    }
-
-    my $obj = $popular{$new_params->{epoch}};
+use constant EPOCH_RE         => qr/^-?[0-9]{1,13}$/o;
+use constant EPOCH_MAYBE_FRAC => qr/^-?[0-9]{1,13}(?:\.[0-9]+)?$/o;
```
Comment (shaydo-deriv) [inline lib/Date/Utility.pm:411]: where is 13 digits limit comes from? don't care, just curious.

**Review by shaydo-deriv — COMMENTED** [review]
code changes look good to me, just question regarding if LICENSE is needed and if the AUTHOR should be BINARY or DERIV. I assume the upload is going to happen under DERIV account.

File: lib/Date/Utility.pm (hunk 2 above)
Comment (mwm-deriv) [reply lib/Date/Utility.pm:411]: So... it is apparently acceptable to produce a "date string" like `YYYYMMDDHHMMSS` which is 14 digits.  From comments in there,  I believe it also used to be acceptable to produce `YYYYMMDD` so it was clamped at `{9,13}`.

File: LICENSE (hunk 1 above)
Comment (mwm-deriv) [reply LICENSE]: Ok.  Just doing `dzil test` produced the LICENSE, so I added it in.  Easily dropped.

**Comment by mwm-deriv** [comment]
> code changes look good to me, just question regarding if LICENSE is needed and if the AUTHOR should be BINARY or DERIV. I assume the upload is going to happen under DERIV account.

I looked this up on MetaCPAN and it's owned by `BINARY`.  WIki docs don't suggest moving things between which I imagine to be a big hassle.

**Comment by shaydo-deriv** [comment]
> > code changes look good to me, just question regarding if LICENSE is needed and if the AUTHOR should be BINARY or DERIV. I assume the upload is going to happen under DERIV account.
> 
> I looked this up on MetaCPAN and it's owned by `BINARY`. WIki docs don't suggest moving things between which I imagine to be a big hassle.



> > code changes look good to me, just question regarding if LICENSE is needed and if the AUTHOR should be BINARY or DERIV. I assume the upload is going to happen under DERIV account.
> 
> I looked this up on MetaCPAN and it's owned by `BINARY`. WIki docs don't suggest moving things between which I imagine to be a big hassle.

ok, then just LICENSE file question

**Review by chylli-deriv — APPROVED** [review]

**Pull request closed without merging**